src/routes/safe/store/actions/transactions/fetchTransactions/loadAllTransactions.ts

```diff
@@ -0,0 +1,81 @@
+import { List, Map } from 'immutable'
+import axios, { AxiosResponse } from 'axios'
+
+import { getAllTxServiceUriTo, getTxServiceHost } from 'src/config'
+import { checksumAddress } from 'src/utils/checksumAddress'
+import { Transaction } from '../../../models/types/transactions'
+
+type ServiceUriParams = {
+  safeAddress: string
+  limit?: number
+  offset?: number
+  orderBy?: string // todo: maybe this should be key of MultiSigTransaction | keyof EthereumTransaction
+  queued?: boolean
+  trusted?: boolean
+}
+
+type EndpointResponse = {
+  count: number
+  next?: string
+  previous?: string
+  results: Transaction[]
+}
+
+const getAllTransactionsUrl = (safeAddress: string) => {
+  const host = getTxServiceHost()
+  const address = checksumAddress(safeAddress)
+  const base = getAllTxServiceUriTo(address)
+
+  return `${host}${base}`
+}
+
+const fetchAllTransactions = async (
+  urlParams: ServiceUriParams,
+  eTag: string | null,
+): Promise<{ eTag: string; results: Transaction[] }> => {
+  const { safeAddress, limit, offset, orderBy, queued, trusted } = urlParams
+  try {
+    const url = getAllTransactionsUrl(safeAddress)
+
+    const config = {
+      params: {
+        limit,
+        offset,
+        orderBy,
+        queued,
+        trusted,
+      },
+      headers: eTag ? { 'If-None-Match': eTag } : undefined,
+    }
+
+    const response: AxiosResponse<EndpointResponse> = await axios.get(url, config)
+
+    if (response.data.count > 0) {
+      const { etag } = response.headers
+
+      if (eTag !== etag) {
+        return {
+          eTag: etag,
+          results: response.data.results,
+        }
+      }
+    }
+  } catch (err) {
+    if (!(err && err.response && err.response.status === 304)) {
+      console.error(`Requests for outgoing transactions for ${safeAddress || 'unknown'} failed with 404`, err)
```

Comment:
`outgoing` is wrong. Also, if `safeAddress ` is undefined, we shouldn't send the request. We should add that condition. and remove the `or` in the message.

Reply:
not really, now we have it typed and `safeAddress` is mandatory, so it will never be undefined.